src/components/Widgets/Form/AddressSelect.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import shortid from 'shortid';
+import Identicon from '../../Identicon'
+import { combineWallets } from '../../../util/helpers';
+
+export default class AddressSelect extends Component {
+  static displayName = 'AddressSelect';
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    wallets: PropTypes.object,
+    walletContracts: PropTypes.object,
+    identiconSize: PropTypes.string,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    wallets: {},
+    walletContracts: {},
+    identiconSize: 'tiny',
+  };
+
+  state = {
+    selectedWallet: '',
+    combinedWallets: [],
+  }
+
+  componentDidMount() {
+    const { wallets, walletContracts, onChange } = this.props;
+    const combinedWallets = combineWallets(wallets, walletContracts);
+    const address = combinedWallets[0] ? combinedWallets[0].address : ''
+    this.setState(
+      { combinedWallets, selectedWallet: address },
+      onChange(combinedWallets[0])
+    )
+  }
+
+  selectAddress = e => {
+    const { onChange } = this.props;
+    this.setState({ selectedWallet: e.target.value }, onChange(e))
+  }
+
+  render() {
+    const { disabled, identiconSize } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+    return (
+      <StyledDiv>
+        <StyledSelect
+          onChange={this.selectAddress}
+          disabled={disabled}
+          value={selectedWallet}
+        >
+          {combinedWallets.map(w => {
+            return (
+              <StyledOption key={shortid.generate()} value={w.address}>
+                {
+                  w.addressType === 'wallet'
+                    ? `🔑 ${w.name}`
+                    : w['contract-name']
+                }
+                &nbsp; - &nbsp;
+                { w.balance }
+                &nbsp;
+                ETHER
+              </StyledOption>
+            );
+          })}
+        </StyledSelect>
+        <StyledIdenticon
+          title
+          size={identiconSize}
+          address={selectedWallet}
+        />
+        </StyledDiv>
```

Comment:
Style nit: indentation is off here. If you use `prettier` in your editor, this shouldn't be an issue. I've just made (and self-assigned) a new task to add prettier as a commit hook to auto-fix this sort of thing in the future.

Reply:
fyi - just merged a PR containing the prettier hook; please rebase on master. you may need to blow away your `yarn.lock` and do a fresh `yarn`.

Reply:
^^ I will pull.